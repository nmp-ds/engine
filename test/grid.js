import { expect, test } from 'vitest';
import { setup } from './_helpers.js';

setup();

const columns = Array.from({ length: 13 }, (_, index) => 1 + index);
const rows = Array.from({ length: 7 }, (_, index) => 1 + index);

test('grid span', async (t) => {
  const staticClasses = [
    'col-auto',
    'row-auto',
    'col-span-full',
    'row-span-full',
  ];

  const artbitraryClassesRows = rows.map((num) => `row-span-${num}`);
  const artbitraryClassesCols = columns.map((num) => `col-span-${num}`);

  const { css } = await t.uno.generate([
    ...staticClasses,
    ...artbitraryClassesCols,
    ...artbitraryClassesRows,
  ]);

  expect(css).toMatchSnapshot();
});

test('grid starts and ends', async (t) => {
  const classesRows = rows.map((num) => [`row-start-${num}`, `row-end-${num}`]);
  const classesCols = columns.map((num) => [`col-start-${num}`, `col-end-${num}`]);

  const { css } = await t.uno.generate([...classesRows, ...classesCols].flat());
  expect(css).toMatchSnapshot();
});

test('grid template columns with arbitrary values', async (t) => {
  const arbitrary = [
    'grid-cols-[200px_minmax(900px,_1fr)_100px]',
    'grid-cols-[320px_1fr]',
    'grid-cols-minmax-200px',
  ];
  const { css } = await t.uno.generate(arbitrary);
  expect(css).toMatchSnapshot();
});

test('grid template rows with arbitrary values', async (t) => {
  const arbitrary = [
    'grid-rows-[200px_repeat(auto-fill, 100px)_300px]',
    'grid-rows-minmax-400px',
    'grid-rows-minmax-400px',
  ];
  const { css } = await t.uno.generate(arbitrary);
  expect(css).toMatchSnapshot();
});

test('grid template rows and cols with not correct arbitrary values', async (t) => {
  const arbitrary = [
    'grid-cols-200px_minmax(900px,_1fr)_100px',
    'grid-rows-200px_repeat(auto-fill, 100px)_300px',
    'grid-rows-minmax-[200px-400px]',
  ];
  const { css } = await t.uno.generate(arbitrary);
  expect(css).toMatchInlineSnapshot('""');
});

test('grid auto flows', async (t) => {
  const gridFlows = [
    'auto-rows-auto',
    'auto-rows-min',
    'auto-rows-max',
    'auto-rows-fr',
    'auto-cols-auto',
    'auto-cols-min',
    'auto-cols-max',
    'auto-cols-fr',
    'grid-flow-row',
    'grid-flow-col',
    'grid-flow-dense',
    'grid-flow-row-dense',
    'grid-flow-col-dense',
  ];
  const { css } = await t.uno.generate(gridFlows);

  expect(css).toMatchSnapshot();
});

test('grid templates basic', async (t) => {
  const classesRows = rows.map((num) => `grid-rows-${num}`);
  const classesCols = columns.map((num) => `grid-cols-${num}`);
  const { css } = await t.uno.generate([...classesCols, ...classesRows]);

  expect(css).toMatchSnapshot();
});

test('grid templates none', async (t) => {
  const classes = ['grid-rows-none', 'grid-cols-none'];
  const { css } = await t.uno.generate(classes);

  expect(css).toMatchSnapshot();
});
