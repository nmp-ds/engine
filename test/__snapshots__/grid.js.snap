// Vitest Snapshot v1

exports[`grid auto flows 1`] = `
"/* layer: default */
.auto-rows-auto{grid-auto-rows:auto;}
.auto-rows-fr{grid-auto-rows:minmax(0,1fr);}
.auto-rows-max{grid-auto-rows:max-content;}
.auto-rows-min{grid-auto-rows:min-content;}
.auto-cols-auto{grid-auto-columns:auto;}
.auto-cols-fr{grid-auto-columns:minmax(0,1fr);}
.auto-cols-max{grid-auto-columns:max-content;}
.auto-cols-min{grid-auto-columns:min-content;}
.grid-flow-row{grid-auto-flow:row;}
.grid-flow-col{grid-auto-flow:column;}
.grid-flow-dense{grid-auto-flow:dense;}
.grid-flow-row-dense{grid-auto-flow:row dense;}
.grid-flow-col-dense{grid-auto-flow:column dense;}"
`;

exports[`grid span 1`] = `
"/* layer: default */
.col-auto{grid-column:auto;}
.row-auto{grid-row:auto;}
.col-span-full{grid-column:1 / -1;}
.row-span-full{grid-row:1 / -1;}
.row-span-1{grid-row:span 1/span 1;}
.row-span-2{grid-row:span 2/span 2;}
.row-span-3{grid-row:span 3/span 3;}
.row-span-4{grid-row:span 4/span 4;}
.row-span-5{grid-row:span 5/span 5;}
.row-span-6{grid-row:span 6/span 6;}
.row-span-7{grid-row:span 7/span 7;}
.col-span-1{grid-column:span 1/span 1;}
.col-span-10{grid-column:span 10/span 10;}
.col-span-11{grid-column:span 11/span 11;}
.col-span-12{grid-column:span 12/span 12;}
.col-span-13{grid-column:span 13/span 13;}
.col-span-2{grid-column:span 2/span 2;}
.col-span-3{grid-column:span 3/span 3;}
.col-span-4{grid-column:span 4/span 4;}
.col-span-5{grid-column:span 5/span 5;}
.col-span-6{grid-column:span 6/span 6;}
.col-span-7{grid-column:span 7/span 7;}
.col-span-8{grid-column:span 8/span 8;}
.col-span-9{grid-column:span 9/span 9;}"
`;

exports[`grid starts and ends 1`] = `
"/* layer: default */
.row-start-1{grid-row-start:1;}
.row-start-2{grid-row-start:2;}
.row-start-3{grid-row-start:3;}
.row-start-4{grid-row-start:4;}
.row-start-5{grid-row-start:5;}
.row-start-6{grid-row-start:6;}
.row-start-7{grid-row-start:7;}
.col-start-1{grid-column-start:1;}
.col-start-10{grid-column-start:10;}
.col-start-11{grid-column-start:11;}
.col-start-12{grid-column-start:12;}
.col-start-13{grid-column-start:13;}
.col-start-2{grid-column-start:2;}
.col-start-3{grid-column-start:3;}
.col-start-4{grid-column-start:4;}
.col-start-5{grid-column-start:5;}
.col-start-6{grid-column-start:6;}
.col-start-7{grid-column-start:7;}
.col-start-8{grid-column-start:8;}
.col-start-9{grid-column-start:9;}
.row-end-1{grid-row-end:1;}
.row-end-2{grid-row-end:2;}
.row-end-3{grid-row-end:3;}
.row-end-4{grid-row-end:4;}
.row-end-5{grid-row-end:5;}
.row-end-6{grid-row-end:6;}
.row-end-7{grid-row-end:7;}
.col-end-1{grid-column-end:1;}
.col-end-10{grid-column-end:10;}
.col-end-11{grid-column-end:11;}
.col-end-12{grid-column-end:12;}
.col-end-13{grid-column-end:13;}
.col-end-2{grid-column-end:2;}
.col-end-3{grid-column-end:3;}
.col-end-4{grid-column-end:4;}
.col-end-5{grid-column-end:5;}
.col-end-6{grid-column-end:6;}
.col-end-7{grid-column-end:7;}
.col-end-8{grid-column-end:8;}
.col-end-9{grid-column-end:9;}"
`;

exports[`grid template columns with arbitrary values 1`] = `
"/* layer: default */
.grid-cols-\\\\[200px_minmax\\\\(900px\\\\,_1fr\\\\)_100px\\\\]{grid-template-columns:200px minmax(900px, 1fr) 100px;}
.grid-cols-\\\\[320px_1fr\\\\]{grid-template-columns:320px 1fr;}
.grid-cols-minmax-200px{grid-template-columns:repeat(auto-fill,minmax(200px,1fr));}"
`;

exports[`grid template rows with arbitrary values 1`] = `
"/* layer: default */
.grid-rows-\\\\[200px_repeat\\\\(auto-fill\\\\,\\\\ 100px\\\\)_300px\\\\]{grid-template-rows:200px repeat(auto-fill, 100px) 300px;}
.grid-rows-minmax-400px{grid-template-rows:repeat(auto-fill,minmax(400px,1fr));}"
`;

exports[`grid templates basic 1`] = `
"/* layer: default */
.grid-rows-1{grid-template-rows:repeat(1,minmax(0,1fr));}
.grid-rows-2{grid-template-rows:repeat(2,minmax(0,1fr));}
.grid-rows-3{grid-template-rows:repeat(3,minmax(0,1fr));}
.grid-rows-4{grid-template-rows:repeat(4,minmax(0,1fr));}
.grid-rows-5{grid-template-rows:repeat(5,minmax(0,1fr));}
.grid-rows-6{grid-template-rows:repeat(6,minmax(0,1fr));}
.grid-rows-7{grid-template-rows:repeat(7,minmax(0,1fr));}
.grid-cols-1{grid-template-columns:repeat(1,minmax(0,1fr));}
.grid-cols-10{grid-template-columns:repeat(10,minmax(0,1fr));}
.grid-cols-11{grid-template-columns:repeat(11,minmax(0,1fr));}
.grid-cols-12{grid-template-columns:repeat(12,minmax(0,1fr));}
.grid-cols-13{grid-template-columns:repeat(13,minmax(0,1fr));}
.grid-cols-2{grid-template-columns:repeat(2,minmax(0,1fr));}
.grid-cols-3{grid-template-columns:repeat(3,minmax(0,1fr));}
.grid-cols-4{grid-template-columns:repeat(4,minmax(0,1fr));}
.grid-cols-5{grid-template-columns:repeat(5,minmax(0,1fr));}
.grid-cols-6{grid-template-columns:repeat(6,minmax(0,1fr));}
.grid-cols-7{grid-template-columns:repeat(7,minmax(0,1fr));}
.grid-cols-8{grid-template-columns:repeat(8,minmax(0,1fr));}
.grid-cols-9{grid-template-columns:repeat(9,minmax(0,1fr));}"
`;

exports[`grid templates none 1`] = `
"/* layer: default */
.grid-rows-none{grid-template-rows:none;}
.grid-cols-none{grid-template-columns:none;}"
`;
